File: Lista de Exercícios/Python/13. Métodos Estáticos/MetodosEstaticos.py
class Matematica:
    @staticmethod
    def calculaFatorial(numero):
        if numero < 0:
            raise ValueError("O número deve ser positivo")
        if numero == 0 or numero == 1:
            return 1
        resultadoFatorial = 1
        for i in range(1, numero + 1):
            resultadoFatorial *= i
        return resultadoFatorial

File: Lista de Exercícios/Go/13. Métodos Estáticos/MetodosEstaticos.go
package main

import (
	"errors"
	"fmt"
)

func calculaFatorial(numero int) (int, error) {
	if numero < 0 {
		return 0, errors.New("o número deve ser positivo")
	}
	if numero == 0 || numero == 1 {
		return 1, nil
	}
	resultadoFatorial := 1
	for i := 1; i <= numero; i++ {
		resultadoFatorial *= i
	}
	return resultadoFatorial, nil
}

func main() {
	
}
